web/src/user_events.ts

```diff
@@ -131,6 +135,11 @@ export const update_person = function update(event: UserUpdate): void {
             settings_realm_user_settings_defaults.maybe_disable_widgets();
             settings_account.update_account_settings_display();
             settings.update_lock_icon_in_sidebar();
+            settings_account.update_user_own_role_dropdown_state();
+        }
+
+        if (!people.is_my_user_id(event.user_id) && was_owner !== user.is_owner) {
```

Comment:
This should also have a condition `current_user.is_owner` since the dropdown update is required when other user's role is changed only for owners.